### PR TITLE
add hydra.sweeper.params to optuna

### PR DESCRIPTION
**Pull request by jieru-hu**
HPO config standardization #1890

### CODE REVIEW

**Review by pixelb — APPROVED** [review]
Both code and docs look good. thanks

**Review by Jasha10 — APPROVED** [review]
Nice!

File: website/docs/upgrades/1.1_to_1.2/changes_to_sweeper_config.md [hunk 1]
```diff
@@ -0,0 +1,58 @@
+---
+id: changes_to_sweeper_config
+title: Changes to configuring sweeper's search space
+hide_title: true
+---
+
+Hydra 1.2 introduces `hydra.sweeper.params`. All Hydra Sweepers (BasicSweeper and HPOs) search
+space will be defined under this config node.
```
[the remark this answers is not in the document]
Comment (jieru-hu) [reply website/docs/upgrades/1.1_to_1.2/changes_to_sweeper_config.md:8]: Thanks! The `search_space` config only applies to optuna so probably should leave that out in this overview part of the doc.

File: plugins/hydra_optuna_sweeper/hydra_plugins/hydra_optuna_sweeper/_impl.py [hunk 2]
```diff
@@ -116,22 +125,53 @@ def __init__(
...
+                    "Both hydra.sweeper.params and hydra.sweeper.search_space are configured."
+                    "\nHydra will use hydra.sweeper.params for defining search space."
+                    f"\n{url}"
+                )
+                self.search_space = OmegaConf.create({})
+            else:
+                deprecation_warning(
+                    message=dedent(
+                        f"""\
+                        `hydra.sweeper.searchspace` is deprecated and will be removed in the next major release.
```
[the remark this answers is not in the document]
Comment (jieru-hu) [reply plugins/hydra_optuna_sweeper/hydra_plugins/hydra_optuna_sweeper/_impl.py]: nice catch! thanks!

File: website/docs/upgrades/1.1_to_1.2/changes_to_sweeper_config.md (hunk 1 above)
[the remark this answers is not in the document]
Comment (Jasha10) [reply website/docs/upgrades/1.1_to_1.2/changes_to_sweeper_config.md:8]: Oh, got it. That makes sense.

**Pull request merged**